rest/server_context_test.go

```diff
@@ -163,3 +164,156 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Async:         true,
+		Proxy:         "http://127.0.0.1:443/proxy",
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d63"})
+
+	// Start a continuous fake replication; it should be added to active tasks
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Continuous:    true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d64"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	serverContext.startReplicators()
+
+	activeTasks := serverContext.replicator.ActiveTasks()
+	assert.Equal(t, 1, len(activeTasks))
+	activeTask := activeTasks[0]
+	assert.Equal(t, "58a632bb8d7e110445d3d65a98365d64", activeTask.ReplicationID)
+	serverContext.Close()
+}
+
+func TestPostStartup(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate)()
+	var replications []*ReplicationConfig
+
+	// Start a continuous fake replication.
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Continuous:    true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d62"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	serverContext.startReplicators()
+
+	activeTasks := serverContext.replicator.ActiveTasks()
+	assert.Equal(t, 1, len(activeTasks))
+	activeTask := activeTasks[0]
+	assert.Equal(t, "58a632bb8d7e110445d3d65a98365d62", activeTask.ReplicationID)
+	serverContext.Close()
+}
+
+func TestGetOrAddDatabaseFromConfig(t *testing.T) {
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	oldRevExpirySeconds := uint32(600)
+	localDocExpirySecs := uint32(60 * 60 * 24 * 10) // 10 days in seconds
+
+	// Get or add database name from config without valid database name; throws 400 Illegal database name error
+	dbConfig := &DbConfig{OldRevExpirySeconds: &oldRevExpirySeconds, LocalDocExpirySecs: &localDocExpirySecs}
+	dbContext, err := serverContext._getOrAddDatabaseFromConfig(dbConfig, false)
+	assert.Error(t, err, "It should throw 400 Illegal database name")
+	assert.Contains(t, err.Error(), strconv.Itoa(http.StatusBadRequest))
+
+	// Get or add database from config with unrecognized value for import_docs.
+	dbConfig = &DbConfig{
+		Name:                "imdb",
+		OldRevExpirySeconds: &oldRevExpirySeconds,
+		LocalDocExpirySecs:  &localDocExpirySecs,
+		AutoImport:          "Unknown"}
+
+	dbContext, err = serverContext._getOrAddDatabaseFromConfig(dbConfig, false)
```

Comment:
assert nil on dbContext

Reply:
Asserted nil on dbContext while trying to get or add database from config with unrecognized value for import_docs.